lib/Connection.php

```diff
@@ -249,7 +256,7 @@ private function closeSocket() {
         $this->outputBuffer = "";
         $this->outputBufferLength = 0;
 
-        if (is_resource($this->socket)) {
+        if (!$this->persistent && is_resource($this->socket)) {
```

Comment:
Better just skip the `$this->closeSocket();` call in `__destruct` when the connection is persistent. Also: How do we handle edge cases where a connection isn't in a clean state when exiting? E.g. still subscribed to a channel or awaiting something when `blpop` is used?